examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst

```diff
@@ -0,0 +1,117 @@
+
+This guide demonstrates how to use ``CMakeToolchain`` with predefined generators like ``Ninja`` and how to configure it to use different generators.
+
+Creating a Basic Project
+-------------------------
+
+We will create a basic project based on the ``cmake_exe`` template as an example of a C++ project:
+
+.. code-block:: bash
+
+    $ conan new -d name=foo -d version=0.1.0 cmake_exe
+
+Understanding CMake Generators
+------------------------------
+
+The `CMake <https://cmake.org/>`_ client offers a variety of `generators <https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html>`_ to create build system files.
+If you want to use a generator other than the default chosen by CMake, you can configure ``tools.cmake.cmaketoolchain:generator``.
+
+Please, note that `CMake <https://cmake.org/>`_ client is not the same as the Conan :ref:`CMake <conan_tools_cmake_helper>` helper.
+
+To see which generators are available on your system, run:
+
+.. code-block:: bash
+
+    $ cmake --help
+
+You can set this configuration in your profile, directly in the command line, or even in your global configuration.
+
+Generating the Toolchain File with the Ninja Generator
+------------------------------------------------------
+
+The project's recipe declares the "CMakeToolchain" generator.
+
+1. **Default Build**: To build the recipe using the default generator chosen by CMake, run:
+
+    .. code-block:: bash
+
+        $ conan create .
+
+2. **Specifying the Ninja Generator**: If you prefer to use the Ninja generator, specify it in the command line:
+
+    .. code-block:: bash
+
+        $ conan create . -c tools.cmake.cmaketoolchain:generator=Ninja
+
+    This configuration will be passed to the ``conan_toolchain.cmake`` file, and the Ninja generator will be used.
+    You should see the following output snippet indicating the Ninja generator is being used:
+
+    .. code-block:: bash
+
+        Profile host:
+        [settings]
+        ...
+        [conf]
+        tools.cmake.cmaketoolchain:generator=Ninja
+
+        ...
+        foo/0.1.0: Calling build()
+        foo/0.1.0: Running CMake.configure()
+        foo/0.1.0: RUN: cmake -G "Ninja" ...
+
+Using the Ninja Generator by Default in a Profile
+-------------------------------------------------
+
+To set the Ninja generator as the default in a profile,
+add the entry ``[conf]`` to your profile with the generator value:
+
+.. code-block:: text
+
+    [settings]
+    os=Linux
+    arch=x86_64
+    compiler=gcc
+    compiler.version=13
+    compiler.libcxx=libstdc++11
+    compiler.cppstd=20
+    build_type=Release
+
+    [conf]
+    *:tools.cmake.cmaketoolchain:generator=Ninja
+
+Now, you can build your project without specifying the generator in the command line:
+
+.. code-block:: bash
+
+    $ conan create .
+
+The Ninja generator will be used by default when running CMake with this profile.
+
+Setting the Ninja Generator Globally
```

Comment:
I think, that instead of having a section for each way of using this configuration, I would pick the one we think it's more appropriate for this conf (maybe setting using a profile) and then just mentioning or just redirecting to the docs on other sections...

Reply:
Right, let's add to profile as example and add other as reference.

Plus, the CMake helper supports hardcoded generator.